src/components/select-custom/select-custom.component.ts

```diff
@@ -0,0 +1,128 @@
+import { ChangeDetectionStrategy, Component, ContentChild, ElementRef, EventEmitter, forwardRef, Input, OnChanges, OnDestroy, OnInit, Output, SimpleChanges, TemplateRef, ViewChild } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'ux-select-custom',
+    templateUrl: './select-custom.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => SelectCustomComponent)
+        }
+    ],
+    host: {
+        '[style.width.px]': 'width'
+    }
+})
+export class SelectCustomComponent<T> implements ControlValueAccessor, OnChanges, OnInit, OnDestroy {
+
+    /** Define the selected item */
+    @Input() selected: T;
+
+    /** Hide the filter input */
+    @Input() hideFilter: boolean;
+
+    /** Debounce the filter change event emitter */
+    @Input() filterChangeDebounce: number = 500;
+
+    /** Define the width of the component */
+    @Input() width: number;
+
+    /** Define the max height of the dropdown */
+    @Input() maxHeight: number;
+
+    /** Determine if we should show the loading indicator */
+    @Input() busy: boolean;
+
+    /** Define if null values are allowed */
+    @Input() allowNull: boolean;
+
+    /** Define the placeholder for the filter input */
+    @Input() placeholder: string = 'Type to filter...';
+
+    /** Emit when the selected item is changed */
+    @Output() selectedChange = new EventEmitter<T>();
+
+    /** Emit when the filter text is changed */
+    @Output() filterChange = new EventEmitter<string>();
+
+    @ContentChild('dropdownContent', { static: false }) dropdownContentRef: TemplateRef<void>;
+    @ContentChild('buttonContent', { static: false }) buttonContentRef: TemplateRef<void>;
```

Comment:
I see. I am fine with changing it. How about "displayContent"? Or "toggleContent" / "dropdownToggleContent"?

Reply:
@donna-marijne Any suggestions or preference?

Reply:
I'd be in favor of making this the default content template, since it's the main view of the component.

Reply:
I would like to keep the two ng-templates (see my reasoning discussion above).

Reply:
The reason it's hard to name is because it's the main presentation of the component, hence my suggestion.

Reply:
I do not think it is hard to name - "displayContent" sounds fine for me. Since the first name was obviously confusing, I wanted to double-check and asked for feedback.

Reply:
I think that this should be the default template, and not require a name.

Reply:
I didn't really see the technical reason why the component shouldn't have a default template, but perhaps I missed something?

Reply:
I think technically, both solutions are possible. I also don't see a technical reason for not staying with two named templates.

It is more a logical reason. From the discussion above:

> We have two templates: one might usually be bigger, but the other one is equally important. They work side-by-side and their implementations need to be connected. So they have an equal weight in the implementation. Putting both inside ng-templates stresses this "equality".

I do not have such a strong opinion, but since Ashley prefers to make the other template the default template, I am not sure how we proceed here.

Reply:
I do agree with Donna, that we do usually try to provide a sensible default template, and if this is something that might need to be customized eg, for localization purposes, then we can allow an `ng-template` to customize it but not require it, so having a sensible default makes it easiest for consumers to start using the component. This is the way our other components tend to work.

In regards to using `ng-content` there is a technical reason why we should choose `ng-content` over `ng-template` for the dropdown content. We normally use `ng-template`s to customize a part of a component (such as the button content), but for content that is essentially a part of the page component it is on, `ng-content` should be used. This is the case for this component, where the contents in the dropdown may bind to a `FormGroup` or be using `ngModel`, binding to values on the page it is on, not values related to the select component.

This is important because when change detection is run will differ. `ng-content` runs change detection when the page component changes, whereas `ng-template` runs change detection when the `select-custom` component updates. We can see the difference if we look at the Angular compiler output (Showing Ivy output as it is easier to read):

```html
<ux-select-custom>
  {{ text }}

  <ng-template #customTemplate>
    {{ text }}
  </ng-template>
</ux-select-custom>
```

Would compile to:

#### App Component
```typescript
template: function AppComponent_Template(rf, ctx) {
    if (rf & 1) { // create step
        i0.ɵɵelementStart(0, "ux-select-custom");
        i0.ɵɵtext(1); // ng-content inlined and part of app component
        i0.ɵɵtemplate(2, AppComponent_ng_template_2_Template, 1, 1, "ng-template", null, _c0, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵelementEnd();
    }
    if (rf & 2) { // update step - run when changes occur in app component
        i0.ɵɵselect(1);
        i0.ɵɵtextInterpolate1(" ", ctx.text, " "); // update ng-content if a change occurs
    }
},
```

#### Select Component

```typescript
template: function ChildComponent_Template(rf, ctx) {
    if (rf & 1) { // create step
        i0.ɵɵprojectionDef();
        i0.ɵɵprojection(0);
        i0.ɵɵelementContainerStart(1, _c1);
        i0.ɵɵelementContainerEnd();
    }
    if (rf & 2) { // update step - run when changes occur in select component
        i0.ɵɵselect(1);
        i0.ɵɵproperty("ngTemplateOutlet", ctx.template); // template updated when changes occur in this component
    }
},
```

I hope this makes sense.

Reply:
Thank you very much for the clarification and examples, this helps a lot. I do think that binding dropdown content to the page's FormGroup is unlikely in practice, but there was an example with number pickers where one would probably bind the `ngModel`s to the page.

Reply:
I changed the name to displayContent and created a default template for it.